From 8832e6dc15ac5886ca5dfd6034feb2c3537b257c Mon Sep 17 00:00:00 2001
From: Viswanathan M <viswanathan.m@adlinktech.com>
Date: Mon, 21 Sep 2020 15:01:25 +0530
Subject: [PATCH 5/7] IMX8QM: Modified Changes for SPIFlash MX25UM U5

This patch contains modifications for the earlier patch related to SPI Flash.

Signed-off-by: Viswanathan M <viswanathan.m@adlinktech.com>
---
 arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi | 3 +--
 drivers/mtd/spi-nor/spi-nor.c                     | 2 +-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi b/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi
index 9a1af0e..d5e5bb2 100644
--- a/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm.dtsi
@@ -801,9 +801,8 @@
 		reg = <0>;
 		#address-cells = <1>;
 		#size-cells = <1>;
-		compatible = "micron,mx25um51345g";
+		compatible = "macronix,mx25um51345g";
 		spi-max-frequency = <133000000>;
-		/*spi-nor,ddr-quad-read-dummy = <8>;*/
 	};
 };
 
diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 57939d0..25f3120 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -1115,7 +1115,7 @@ static const struct flash_info spi_nor_ids[] = {
 	{ "mx66u51235f", INFO(0xc2253a, 0, 64 * 1024, 1024, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
 	{ "mx66l1g45g",  INFO(0xc2201b, 0, 64 * 1024, 2048, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ "mx66l1g55g",  INFO(0xc2261b, 0, 64 * 1024, 2048, SPI_NOR_QUAD_READ) },
-	{ "mx25um51345g",INFO(0xc2813a, 0, 64 * 1024, 1024, SECT_4K | SPI_NOR_4B_OPCODES)}, /*Adlink: Added to support macronix spi flash*/
+	{ "mx25um51345g",INFO(0xc2813a, 0, 64 * 1024, 1024, SECT_4K)}, /*Adlink: Added to support macronix spi flash*/
 
 	/* Micron */
 	{ "n25q016a",	 INFO(0x20bb15, 0, 64 * 1024,   32, SECT_4K | SPI_NOR_QUAD_READ) },
-- 
2.10.2

